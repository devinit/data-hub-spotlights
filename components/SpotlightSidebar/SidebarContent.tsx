import React, { FunctionComponent } from 'react';

interface SidebarContentProps {
  height?: string;
}

const SidebarContent: FunctionComponent<SidebarContentProps> = ({ children, height }) => {
  return (
    <div className="spotlight__content" style={{ height }}>
      {children}
      <style jsx>{`
        height: 100%;
      `}</style>
    </div>
  );
};

export { SidebarContent };
