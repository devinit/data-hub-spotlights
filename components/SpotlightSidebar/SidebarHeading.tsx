import React, { FunctionComponent } from 'react';

interface SidebarHeadingProps {
  heading?: string;
  onClick?: () => void;
}

const SidebarHeading: FunctionComponent<SidebarHeadingProps> = ({ heading, onClick }) => {
  return (
    <h2 className="spotlight__heading">
      { heading }
      <button data-testid="spotlight-menu-trigger" type="button" className="button button--minor" onClick={ onClick }>
        <i role="presentation" aria-hidden="true" className="ico ico--20 ico-arrow-down-blank"/>
      </button>
    </h2>
  );
};

export { SidebarHeading };
