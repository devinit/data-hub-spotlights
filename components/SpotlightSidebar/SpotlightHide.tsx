import React, { FunctionComponent } from 'react';

interface SpotlightHideProps {
  height?: string;
}

const SpotlightHide: FunctionComponent<SpotlightHideProps> = ({ children, height }) => {
  return (
    <div className="spotlight__hide-ss" style={{ height }}>
      {children}
      <style jsx>{`
        height: 100%;
      `}</style>
    </div>
  );
};

export { SpotlightHide };
