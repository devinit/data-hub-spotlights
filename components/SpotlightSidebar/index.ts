export * from './SpotlightSidebar';
export * from './SidebarContent';
export * from './SidebarHeading';
