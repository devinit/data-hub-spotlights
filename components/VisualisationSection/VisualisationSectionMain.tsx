import React, { FunctionComponent } from 'react';

const VisualisationSectionMain: FunctionComponent = ({ children }) => (
  <div className="spotlight__main">
    {children}
    <style jsx>{`
      .spotlight__main :global(.leaflet-container) {
        background: inherit;
      }
    `}</style>
  </div>
);

export { VisualisationSectionMain };
