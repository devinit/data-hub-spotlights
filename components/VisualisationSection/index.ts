export * from './VisualisationSection';
export * from './VisualisationSectionMain';
