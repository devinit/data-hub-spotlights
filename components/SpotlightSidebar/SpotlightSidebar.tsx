import React, { Children, FunctionComponent, isValidElement } from 'react';
import { SidebarContent } from './SidebarContent';
import { SidebarHeading } from './SidebarHeading';

const SpotlightSidebar: FunctionComponent = ({ children }) => {

  return (
    <div className="spotlight__aside spotlight__aside--padded">
      {
        Children.map(children, child =>
          isValidElement(child) && (child.type === SidebarContent || child.type === SidebarHeading) ? child : null)
      }
      <style jsx>{ `
        min-height: 500px;
      ` }</style>
    </div>
  );
};

export { SpotlightSidebar };
