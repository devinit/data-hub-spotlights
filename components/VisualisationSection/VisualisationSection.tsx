import classNames from 'classnames';
import React, { Children, FunctionComponent, isValidElement, ReactNode } from 'react';
import { SpotlightSidebar } from '../SpotlightSidebar';
import { VisualisationSectionMain } from './VisualisationSectionMain';

const VisualisationSection: FunctionComponent<{ className?: string }> = ({ children, className }) => {
  const renderContent = (): ReactNode =>
    Children.map(children, child => {
      if (isValidElement(child) && (child.type === SpotlightSidebar || child.type === VisualisationSectionMain)) {
        return child;
      }
    });

  return <div className={classNames('spotlight', className)}>{renderContent()}</div>;
};

export { VisualisationSection };
