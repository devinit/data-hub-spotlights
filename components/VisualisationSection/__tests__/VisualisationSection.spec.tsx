/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SpotlightSidebar } from '../../SpotlightSidebar';
import { VisualisationSection } from '../VisualisationSection';
import { VisualisationSectionMain } from '../VisualisationSectionMain';

describe('VisualisationSection', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<VisualisationSection />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders only SpotlightSidebar & VisualisationSectionMain', () => {
    const renderer = TestRenderer.create(
      <VisualisationSection>
        <SpotlightSidebar />
        <VisualisationSectionMain />
        <div>This will not Show</div>
      </VisualisationSection>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with the specified class name', () => {
    const renderer = TestRenderer.create(<VisualisationSection className="spotlight--leader" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
