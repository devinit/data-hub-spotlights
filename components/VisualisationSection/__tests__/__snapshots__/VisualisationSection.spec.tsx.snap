// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualisationSection renders correctly 1`] = `
<div
  className="spotlight"
/>
`;

exports[`VisualisationSection renders only SpotlightSidebar & VisualisationSectionMain 1`] = `
<div
  className="spotlight"
>
  <div
    className="spotlight__aside"
  >
    <style
      jsx={true}
    >
      
        min-height: 500px;
      
    </style>
  </div>
  <div
    className="spotlight__main"
  >
    <style
      jsx={true}
    >
      
      min-height: 500px;
      border: 1px solid #e84439;

      .spotlight__main :global(.leaflet-container) {
        background: inherit;
      }
    
    </style>
  </div>
</div>
`;

exports[`VisualisationSection renders with the specified class name 1`] = `
<div
  className="spotlight spotlight--leader"
/>
`;
